core/src/main/kotlin/fr/sncf/osrd/stdcm/preprocessing/implementation/BlockAvailability.kt

```diff
@@ -181,7 +183,10 @@ data class BlockAvailability(
                         if (plannedMinTimeAtStep > timeAtStep) {
                             // Train passes through planned timing data before it is available
                             return AvailabilityProperties(
-                                plannedMinTimeAtStep - timeAtStep,
+                                max(
+                                    plannedMinTimeAtStep - timeAtStep - internalMarginForSteps,
+                                    0.0
+                                ),
```

Comment:
I'd prefer if we update `plannedMinTimeAtStep` instead of offsetting it here. It *feels* like we'd have weird edge cases otherwise, like we'd consider that we're too early/late but then actually not